packages/cli/src/queries/system.ts

```diff
@@ -0,0 +1,67 @@
+import type { AlterationState, System } from '@logto/schemas';
+import { systemGuards, Systems, AlterationStateKey } from '@logto/schemas';
+import { convertToIdentifiers } from '@logto/shared';
+import type { Nullable } from '@silverhand/essentials';
+import type { CommonQueryMethods, DatabaseTransactionConnection } from 'slonik';
+import { sql } from 'slonik';
+import { z } from 'zod';
+
+const { fields } = convertToIdentifiers(Systems);
+
+const doesTableExist = async (pool: CommonQueryMethods, table: string) => {
+  const { rows } = await pool.query<{ regclass: Nullable<string> }>(
+    sql`select to_regclass(${table}) as regclass`
+  );
+
+  return Boolean(rows[0]?.regclass);
+};
+
+export const doesSystemsTableExist = async (pool: CommonQueryMethods) =>
+  doesTableExist(pool, Systems.table);
+
+const getAlterationStateTable = async (pool: CommonQueryMethods) =>
+  (await doesSystemsTableExist(pool))
+    ? sql.identifier([Systems.table])
+    : sql.identifier(['_logto_configs']); // Fall back to the old config table
```

Comment:
nit: Could have a const variable defined like `legacyLogtoConfigsTable` somewhere. As after this alteration, the reference to this old table name will no longer exist.  It could provide better context compared to the pain string here.

Reply:
you are right